src/wyzeapy/wyze_auth_lib.py

```diff
@@ -132,21 +148,24 @@ async def get_token_with_2fa(self, verification_code) -> Token:
             'https://auth-prod.api.wyze.com/user/login',
             headers=headers, json=payload)
 
-        self.token = Token(response_json['access_token'], response_json['refresh_token'], time.time(), True)
+        self.token = Token(response_json['access_token'], response_json['refresh_token'])
+        self.token_callback(self.token)
         return self.token
 
     @property
-    def should_refresh(self) -> bool:
-        return time.time() - self.token.last_login_time > 59 * 60 * 60
+    async def should_refresh(self) -> bool:
+        return time.time() >= self.token.refresh_time
 
     async def refresh_if_should(self):
         if self.should_refresh:
-            _LOGGER.debug("Should refresh. Refreshing...")
-            try:
-                await self.refresh()
-            except AccessTokenError:
-                _LOGGER.warning("Could not refresh. Logging in with the Username and Password...")
-                self.token = await self.get_token_with_username_password(self._username, self._password)
+            async with self.refresh_lock:
```

Comment:
Can we move this to the top level instead of doing the if statement twice?

Reply:
I think that this may have been an older comment....